### ISSUE TITLE
Error on Rollback on a workload

### ISSUE THREAD

**Issue by sowmyav27** [rancher/rancher#23269]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
1. Deploy a workload with image: `sangeetha/mytestcontainer`. Scale up the pods to 2.
2. Edit the workload, change image to `ubuntu`.
3. Pods will get recreated
4. Edit the workload, change image to `nginx`.
5. Pods will get recreated
6. Rollback to the first revision id.

**Expected Result:**
The Rollback should be successful

**Actual Result:**
The rollback throws an error
<img width="1547" alt="Screen Shot 2019-10-07 at 2 41 48 PM" src="https://user-images.githubusercontent.com/26032343/66350991-8a913a00-e911-11e9-891a-63e159d6503c.png">

**Note:**
This does NOT happen always.

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): rancher:v2.3.0-rc12
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): custom
- Kubernetes version (use `kubectl version`):

```
1.15
```

- Docker version (use `docker version`):

```
18.9.9
```

**Comment by bmdepesa** [comment]
I was not able to reproduce this at all in a single node setup with a `1.15.4` rke cluster (ec2). 

On @sowmyav27's setup, I noticed I could sometimes retry the rollback on the same workload and it would proceed. Then upon trying to rollback to another revision of the same workload, I would get the error again.

**Comment by bmdepesa** [comment]
Also reproduced on `1.14.7` cluster in @sowmyav27's HA setup.

**Comment by jiaqiluo** [comment]
The bug is reproduced in a fresh installed rancher v2.3.0-rc12 HA mode in the following two clusters
```
cluster 1: k8s 1.15.4, 1 etcd + control plane, 3 workers
cluster 2: k8s v1.14.7, single node 
```

**Comment by jiaqiluo** [comment]
The bug fix is validated in fresh installed v2.3.0-rc13 HA setup, 
```
rancher v2.3.0-rc13, HA install
cluster1 k8s v1.15.4, 1 etcd + control plane, 2 workers
clusetr2 k8s v1.14.7, 1 etcd, 1 control plane, 3 workers
```
The steps are the same as the issue provides,  I could roll back the workload to any revision, and tried multiple workloads multiple times.

**Comment by sowmyav27** [comment]
Verified on `2.3.0-rc13` HA setup.
**Steps:**

- Deploy a workload with image: sangeetha/mytestcontainer. Scale up the pods to 2.
- Edit the workload, change image to ubuntu.
- Pods will get recreated
- Edit the workload, change image to nginx.
- Pods will get recreated
- Rollback to the first revision id.
- Rollback happens successfully. No error

Verified the above set of steps on a couple of workloads.
Cluster spec: 3 node DO rke cluster- 1 etcd, 1 control plane and 1 worker node.
